packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.m

```diff
@@ -0,0 +1,40 @@
+#import "IosPlatformImagesPlugin.h"
+
+#if !__has_feature(objc_arc)
+#error ARC must be enabled!
+#endif
+
+@interface IosPlatformImagesPlugin ()
+@end
+
+@implementation IosPlatformImagesPlugin
+
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
+  FlutterMethodChannel* channel =
+      [FlutterMethodChannel methodChannelWithName:@"plugins.flutter.io/ios_platform_images"
+                                  binaryMessenger:[registrar messenger]];
+
+  [channel setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
+    if ([@"loadImage" isEqualToString:call.method]) {
+      NSString* name = call.arguments;
+      UIImage* image = [UIImage imageNamed:name];
+      NSData* data = UIImagePNGRepresentation(image);
+      if (data) {
+        result([FlutterStandardTypedData typedDataWithBytes:data]);
+      } else {
+        result(nil);
```

Comment:
can we provide users with a bit more details in terms of what went wrong?

Reply:
Returning `nil` is the same contract `[UIImage imageNamed:]` follows.